async-nats/src/lib.rs

```diff
@@ -619,8 +652,20 @@ impl Connector {
                                     // subscription from the map and unsubscribe.
                                     if subscription.sender.send(message).await.is_err() {
                                         context.remove(sid);
-                                        self.connection.write_op(ClientOp::Unsubscribe { id: sid }).await?;
+                                        self.connection.write_op(ClientOp::Unsubscribe { id: sid, max: None }).await?;
                                         self.connection.stream.flush().await?;
+                                    // if channel was open and we sent the messsage, increase the
```

Comment:
We don't care if the channel is open or not, we've received it so in terms of sending the count again to the server this isn't relevant.